hugr-model/src/v0/mod.rs

```diff
@@ -0,0 +1,621 @@
+        func: TermId,
+    },
+    /// Custom operation.
+    ///
+    /// The implicit parameters of the operation are left out.
+    Custom {
+        /// The name of the custom operation.
+        name: GlobalRef<'a>,
+    },
+    /// Custom operation.
+    ///
+    /// The implicit parameters of the operation are included.
+    CustomFull {
+        /// The name of the custom operation.
+        name: GlobalRef<'a>,
+    },
+    /// Alias definitions.
+    DefineAlias {
+        /// The declaration of the alias to be defined.
+        decl: &'a AliasDecl<'a>,
+        /// The value of the alias.
+        value: TermId,
+    },
+
+    /// Alias declarations.
+    DeclareAlias {
+        /// The alias to be declared.
+        decl: &'a AliasDecl<'a>,
+    },
+
+    /// Tail controlled loop.
+    /// Nodes with this operation contain a dataflow graph that is executed in a loop.
+    /// The loop body is executed at least once, producing a result that indicates whether
+    /// to continue the loop or return the result.
+    ///
+    /// # Port Types
+    ///
+    /// - **Inputs**: `inputs` + `rest`
+    /// - **Outputs**: `outputs` + `rest`
+    /// - **Sources**: `inputs` + `rest`
+    /// - **Targets**: `(adt [inputs outputs])` + `rest`
+    TailLoop,
+
+    /// Conditional operation.
+    ///
+    /// # Port types
+    ///
+    /// - **Inputs**: `[(adt inputs)]` + `context`
+    /// - **Outputs**: `outputs`
+    Conditional,
+
+    /// Create an ADT value from a sequence of inputs.
+    Tag {
+        /// The tag of the ADT value.
+        tag: u16,
+    },
+}
+
+/// A region in the hugr.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct Region<'a> {
+    /// The kind of the region. See [`RegionKind`] for details.
+    pub kind: RegionKind,
+    /// The source ports of the region.
+    pub sources: &'a [Port<'a>],
+    /// The target ports of the region.
+    pub targets: &'a [Port<'a>],
+    /// The nodes in the region. The order of the nodes is not significant.
+    pub children: &'a [NodeId],
+    /// The metadata attached to the region.
+    pub meta: &'a [MetaItem<'a>],
+    /// The type of the region.
+    pub r#type: TermId,
+}
+
+/// The kind of a region.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum RegionKind {
+    /// Data flow region.
+    DataFlow,
+    /// Control flow region.
+    ControlFlow,
+}
+
+/// A port attached to a [`Node`] or [`Region`].
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct Port<'a> {
+    /// The link that the port is connected to.
+    pub link: LinkRef<'a>,
+    /// The type of the port.
+    pub r#type: Option<TermId>,
+}
+
+/// A function declaration.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct FuncDecl<'a> {
+    /// The name of the function to be declared.
+    pub name: &'a str,
+    /// The static parameters of the function.
+    pub params: &'a [Param<'a>],
+    /// The type of the function.
+    pub func: TermId,
```

Comment:
```suggestion
    pub f_type: TermId,
```
or some name more specific than "func"?

Reply:
`signature` would work well I think.